packages/npm-dependencies-validation/test/index.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { getDependencyIssues } from "../src/index";
+import { expect, assert } from "chai";
+import * as fsPromises from "fs/promises";
+import { SinonMock, createSandbox } from "sinon";
+
+describe("index unit test", () => {
+  it("2 dependencies and 1 devDependency declared but are not installed", async () => {
+    const result = await getDependencyIssues({
+      fsPath: "./test/projects/no_deps_installed/package.json",
+    });
+    expect(result).to.have.lengthOf(3);
+    // missing json-fixer dependency
+    const jsonFixer = result.find((dep) => dep.name === "json-fixer");
+    assert.isDefined(jsonFixer);
+    expect(jsonFixer?.devDependency).to.be.false;
+    expect(jsonFixer?.type).to.be.equal("missing");
+    expect(jsonFixer?.version).to.be.equal("1.6.12");
+    // missing lodash dependency
+    const lodash = result.find((dep) => dep.name === "lodash");
+    assert.isDefined(lodash);
+    expect(lodash?.devDependency).to.be.false;
+    expect(lodash?.type).to.be.equal("missing");
+    expect(lodash?.version).to.be.equal("~4.17.21");
+    // missing typescript devDependency
+    const typescript = result.find((dep) => dep.name === "typescript");
+    assert.isDefined(typescript);
+    expect(typescript?.devDependency).to.be.true;
+    expect(typescript?.type).to.be.equal("missing");
+    expect(typescript?.version).to.be.equal("^4.4.4");
+  });
+
+  it("1 dependency and 1 devDependency declared but are not installed, 1 dependency is installed", async () => {
+    const result = await getDependencyIssues({
+      fsPath: "./test/projects/some_deps_installed/package.json",
+    });
+    expect(result).to.have.lengthOf(2);
+    // missing lodash dependency
+    const lodash = result.find((dep) => dep.name === "lodash");
+    assert.isDefined(lodash);
+    expect(lodash?.devDependency).to.be.false;
+    expect(lodash?.type).to.be.equal("missing");
+    expect(lodash?.version).to.be.equal("~4.17.21");
+    // missing typescript devDependency
+    const typescript = result.find((dep) => dep.name === "typescript");
+    assert.isDefined(typescript);
+    expect(typescript?.devDependency).to.be.true;
+    expect(typescript?.type).to.be.equal("missing");
+    expect(typescript?.version).to.be.equal("^4.4.4");
+  });
+
+  it("1 dependency and 1 devDependency declared but not installed, 1 dependency is installed but redundant", async () => {
+    const result = await getDependencyIssues({
```

Comment:
I dont see any automation to ensure some of the deps would get installed
how does this work?

Reply:
will checkout to understand this

Reply:
this was the issue with the .gitignore